polar-core/src/resource_block.rs

```diff
@@ -909,7 +1123,7 @@ mod tests {
                 roles=["member"];
                 "member" if "owner";
             }"#,
```

Comment:
Think about edge cases more in these tests:

conflicting declarations (`actor Org`, `resource Org`) - seems like fine from what you said but good to test
conflicting role / permission `roles = ["foo"];`, `permissions = ["foo"];`

Reply:
updated the merging code to only permit overwriting declarations if they are of the same type, otherwise we will raise a validation error.